src/components/checkbox/index.tsx

```diff
@@ -8,43 +8,48 @@ import * as translation from "./translation"
 })
 export class SmoothlyCheckbox {
 	@Element() element: HTMLElement
-	@Prop() selectAll = false
-	@Prop() size: "tiny" | "small" | "medium" | "large" = "small"
-	@Prop({ mutable: true, reflect: true }) intermediate: boolean
-	@Prop({ mutable: true, reflect: true }) selected: boolean
-	@Prop({ mutable: true, reflect: true }) disabled: boolean
-	@Event() checked: EventEmitter<{ selected: boolean }>
+	@Prop() size: "tiny" | "small" | "medium" | "large" = "tiny"
+	@Prop({ mutable: true, reflect: true }) checked = false
+	@Prop({ mutable: true, reflect: true }) intermediate = false
+	@Prop() name: string
+	@Prop() value: any
+	@Prop({ reflect: true }) disabled: boolean
+	@Event() smoothlyChecked: EventEmitter<Record<string, any>>
 	@State() t: langly.Translate
 
 	componentWillLoad() {
 		this.t = translation.create(this.element)
 	}
-	toggle() {
+
+	@Method()
+	async toggle(): Promise<void> {
 		if (!this.disabled) {
-			this.selected = !this.selected
-			this.checked.emit({ selected: this.selected })
+			const checked = this.intermediate || this.checked == false
+			this.smoothlyChecked.emit({
+				[this.name]: checked ? this.value : undefined,
+			})
+			this.checked = checked
 		}
 	}
+
 	render() {
-		return [
-			<smoothly-icon
-				toolTip={this.t(this.selectAll ? "Deselect all" : "Deselect")}
-				onClick={() => this.toggle()}
-				style={{ display: this.selected ? "" : "none" }}
-				size={this.size}
-				name="checkbox-outline"></smoothly-icon>,
-			<smoothly-icon
-				toolTip={this.t(this.intermediate && !this.selected ? "Deselect all" : "Select all")}
-				onClick={() => this.toggle()}
-				style={{ display: this.intermediate && !this.selected ? "" : "none" }}
-				size={this.size}
-				name="remove-outline"></smoothly-icon>,
+		return (
 			<smoothly-icon
-				toolTip={this.t(this.selectAll ? "Select all" : "Select")}
-				onClick={() => this.toggle()}
-				style={{ display: !this.selected ? "" : "none" }}
+				toolTip={this.t(!this.checked ? "Select" : "De-select")}
+				onClick={e => {
+					this.toggle()
+					e.stopPropagation()
```

Comment:
do we want stop propagation on click events? if you have the checkbox in a div and add a click event on the div. the div will not recognize the click event if the checkbox is clicked even if the click event happened inside the div.